packages/cms/lib/modules/resource-form-widgets/lib/api.js

```diff
@@ -0,0 +1,180 @@
+module.exports = async function(self, options) {
+  /**
+   * Sync data with api when user saves or commit
+   * @param req
+   * @param doc
+   * @param options
+   * @returns {Promise<void>}
+   */
+  self.syncConfirmationFields = async (req, doc, options) => {
+    if ((doc.workflowLocale && doc.workflowLocale === 'default-draft') || (!doc.body || !doc.body.items)) {
+      return;
+    }
+
+    doc.body.items.forEach((item) => {
+      const items = self.getAreaItems(item);
+
+      items.forEach(async (item) => {
+        if (item.type === 'resource-form') {
+          if (item.confirmationEnabledUser) {
+            try {
+              await self.addOrUpdateNotification(item, 'User');
+            } catch (error) {
+              console.error(
+                'something went wrong when update admin confirmation settings to api',
+                error.message
+              );
+            }
+          } else {
+            await self.disableNotificationRuleSet(`User-${item.formName}`);
+          }
+
+          if (item.confirmationEnabledAdmin) {
+            try {
+              await self.addOrUpdateNotification(item, 'Admin');
+            } catch (error) {
+              console.error(
+                'something went wrong when update admin confirmation settings to api',
+                error.message
+              );
+            }
+          } else {
+            await self.disableNotificationRuleSet(`Admin-${item.formName}`);
+          }
```

Comment:
Kunnen we deze twee stukken code ook niet in een `for (userType of ['User', 'Admin']) { /* ... */ }` zetten?